### PR TITLE
Feat/Logical object counters

### LINKED ISSUE #1658: Add metric with object count for the node and for each shard.
We need this metric set for testing and for on-prem installations.

### PR DESCRIPTION
**Pull request by carpawell**
Closes #1658.

```
# HELP neofs_node_object_counter Objects counters per shards
# TYPE neofs_node_object_counter gauge
neofs_node_object_counter{shard="H3hDLxLhK78d7Gd4izmsMM",type="logic"} 3
neofs_node_object_counter{shard="H3hDLxLhK78d7Gd4izmsMM",type="phy"} 4
neofs_node_object_counter{shard="RDUb4UwA3rvMbnqEjPZWA8",type="logic"} 1
neofs_node_object_counter{shard="RDUb4UwA3rvMbnqEjPZWA8",type="phy"} 1
```

Will add tests for high-level components. UPD: done.

### CODE REVIEW

File: pkg/local_object_storage/metabase/control.go [hunk 1]
```diff
@@ -128,6 +128,13 @@ func (db *DB) init(reset bool) error {
 	})
 }
 
+// SyncCounters forces to synchronize the object counters.
+func (db *DB) SyncCounters() error {
+	return db.boltDB.Update(func(tx *bbolt.Tx) error {
+		return syncCounter(tx, db.epochState.CurrentEpoch(), true)
```
Comment (fyrchik) [inline pkg/local_object_storage/metabase/control.go:134]: We shouldn't block in an `Update` transaction, could you move epoch fetching out of it?

File: pkg/local_object_storage/metabase/control.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ func (db *DB) init(reset bool) error {
 		}
 
 		if !reset {
-			err = syncCounter(tx)
+			err = syncCounter(tx, db.epochState.CurrentEpoch(), false)
```
Comment (fyrchik) [inline pkg/local_object_storage/metabase/control.go:109]: Same here.

Comment (fyrchik) [reply pkg/local_object_storage/metabase/control.go:109]: Also, I don't see any changes in the `syncCounter` itself in this commit.

File: pkg/local_object_storage/metabase/db_test.go [hunk 3]
```diff
@@ -29,7 +28,7 @@ func (s epochState) CurrentEpoch() uint64 {
 		return s.e
 	}
 
-	return math.MaxUint64
+	return 0
```
Comment (fyrchik) [inline pkg/local_object_storage/metabase/db_test.go:31]: I've thought that default must be a big enough epoch, why do we need this?

File: pkg/local_object_storage/metabase/counters.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	"encoding/binary"
+	"fmt"
+
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.etcd.io/bbolt"
+)
+
+var objectPhyCounterKey = []byte("phy_counter")
+var objectLogicCounterKey = []byte("logic_counter")
```
Comment (fyrchik) [inline pkg/local_object_storage/metabase/counters.go:13]: Can we leave the file in place? It is hard to see the actual diff because of moving.

Comment (fyrchik) [reply pkg/local_object_storage/metabase/counters.go:13]: Also, `VERSION.md` could be updated.

File: pkg/local_object_storage/metabase/counters.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	var phyCounter uint64
+	var logicCounter uint64
+
+	err = iteratePhyObjects(tx, func(cid cid.ID, oid oid.ID) error {
+		phyCounter++
+
+		addr.SetContainer(cid)
+		addr.SetObject(oid)
+
+		if st := objectStatus(tx, addr, epoch); st == 0 || st == 3 {
```
Comment (fyrchik) [inline pkg/local_object_storage/metabase/counters.go:132]: Could you add a short comment about the meaning of `0` and `3`?

File: pkg/local_object_storage/metabase/counters.go (hunk 4 above)
Comment (carpawell) [reply pkg/local_object_storage/metabase/counters.go:13]: > Can we leave the file in place?

ok, renamed it back

> VERSION.md could be updated.

updated

File: pkg/local_object_storage/metabase/control.go (hunk 2 above)
Comment (carpawell) [reply pkg/local_object_storage/metabase/control.go:109]: fixed

File: pkg/local_object_storage/metabase/control.go (hunk 1 above)
Comment (carpawell) [reply pkg/local_object_storage/metabase/control.go:134]: fixed

File: pkg/local_object_storage/metabase/counters.go (hunk 5 above)
Comment (carpawell) [reply pkg/local_object_storage/metabase/counters.go:132]: changed that to `inGraveyardWithKey` func that does not contain `3` result, also, added some comments

File: pkg/local_object_storage/metabase/counter.go [hunk 6]
```diff
@@ -58,41 +99,65 @@ func (db *DB) updateCounter(tx *bbolt.Tx, delta uint64, inc bool) error {
...
-	err = iteratePhyObjects(tx, func(_ cid.ID, _ oid.ID) error {
-		counter++
+	var addr oid.Address
+	var phyCounter uint64
+	var logicCounter uint64
+
+	graveyardBKT := tx.Bucket(graveyardBucketName)
+	garbageBKT := tx.Bucket(garbageBucketName)
+
+	err = iteratePhyObjects(tx, func(cid cid.ID, oid oid.ID) error {
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/metabase/counter.go:129]: Try to not collide with the package names.

**Review by cthulhu-rider — APPROVED** [review]

File: pkg/local_object_storage/metabase/db_test.go (hunk 3 above)
Comment (fyrchik) [reply pkg/local_object_storage/metabase/db_test.go:31]: Shouldn't all tests fail then?
Let it be clear: we use 0 here (epoch source), thus all objects are expired by default. Right?

**Review by cthulhu-rider — DISMISSED** [review]

**Comment by carpawell** [comment]
Also, added tests on the shard level.

**Review by fyrchik — APPROVED** [review]

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**